lib/utils/report.js

```diff
@@ -109,3 +109,32 @@ module.exports = function report(problem) {
 
 	result.warn(warningMessage, warningProperties);
 };
+
+let warningShown = false;
+
+/**
+ * @param {import('postcss').Node | undefined} node
+ * @param {number | undefined} index
+ * @param {number | undefined} endIndex
+ * @returns {number | undefined}
+ */
+function getStartLine(node, index, endIndex) {
+	if (!node) return;
+
+	// HACK: The `rangeBy` method has been added since PostCSS 8.4.0,
+	//       while the `node` may be produced by PostCSS 8.3.x or lower
+	//       that doesn't have the method.
+	//       See https://github.com/stylelint/stylelint/issues/5766
+	if (typeof node.rangeBy !== 'function') {
+		if (!warningShown) {
+			console.warn(
+				'It appears different PostCSS versions are used in your dependency tree. Please consider updating PostCSS to use one version. See https://github.com/stylelint/stylelint/issues/5766 for details.',
+			);
+			warningShown = true;
+		}
+
+		node.rangeBy = postcss.Node.prototype.rangeBy;
```

Comment:
If we ever drop Node.js 12, we could use a dynamic import in the `if`.

Reply:
We don't need this PR if https://github.com/stylelint/stylelint/issues/5766#issuecomment-1207260007 works.